### DISCUSSION TITLE
Query language

### DISCUSSION THREAD

**Discussion by harshad1** [todotxt/todo.txt#66] [General]
Is there a query language / syntax in common use with todo.txt. Something like the [todoist filter language](https://todoist.com/help/articles/introduction-to-filters)?

**Comment by bobwhitelock** [comment]
`grep`?

**Comment by harshad1** [comment]
Lol. I should have been more specific.

I am thinking of implementing a todo.txt query system in [Markor](https://github.com/gsantner/markor). This would be something a user could enter in the app and would be more todo-syntax specific than a general purpose filtering system like grep (or regex).

For example `(*A | *B) & +work` would be +work tasks with priority A or B.

I could come up with something myself, but I'd rather use an existing syntax / format, if it is in use in other clients.

**Comment by harshad1** [comment]
Any opinions on this:

```json
{
    "note": "Todo text filter in json format",
    "and": [
        "project": "work",
        "or" : [
            "context": "meeting",
            "context": "george"
        ],
        "not" : {
            "or": [
                "priority": "C",
                "priority": "D",
                "priority": "E"
            ]
        },
        "or": [
            "due": "today",
            "due": "overdue"
        ],
        "done": false
    ]
}
```
